File: Javascript/Algorithms/1122.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
let relativeSortArray = (arr1, arr2) => {
    // Create mapping for ordering.
    const order = new Map;
    arr2.forEach((value, index) => {
        order.set(value, index);
    });

	// Sorting function (sort based on ordering.)    
    const sorter = (a, b) => {
        a = order.has(a) ? order.get(a) : (a + 1000);
        b = order.has(b) ? order.get(b) : (b + 1000);
        return a - b;
    };
    arr1.sort(sorter);
    return arr1;
};

File: Python/Algorithms/1122.py
class Solution:

    def relativeSortArray(self, arr1, arr2):
        # dictionary holds order which we use to sort arr1.
        order = {j: i for i, j in enumerate(arr2)}
        get = order.get
 
        # use get(x, x+1000), this way when we don't have
        # a value in order for x we can build a sorting
        # relation based on the value of x (+1000 as the
        # max value arr1[i] can have.)
        arr1.sort(key=lambda x: get(x, x+1000))
        return arr1
